File: backend_python/app/models/user_process.py
from sqlalchemy import Column, Integer, DateTime, ForeignKey, Text
from datetime import datetime
from app.models.base import Base

class UserProcess(Base):
    __tablename__ = "user_processes"
    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"))
    process_id = Column(Integer, ForeignKey("processes.id"))
    previous_data = Column(Text(64000))
    updated_data = Column(Text(64000))
    updated_at = Column(DateTime, default=datetime.utcnow)

File: backend_node/app/Models/Sequelize/UserProcess.js
module.exports = (sequelize, DataTypes) => {

    const UserProcess = sequelize.define(
        'user_processes',
        {
            id: {
                type: DataTypes.INTEGER,
                autoIncrement: true,
                primaryKey: true,
                allowNull: false,
            },
            user_id: {
                type: DataTypes.INTEGER,
                allowNull: false
            },
            process_id: {
                type: DataTypes.INTEGER,
                allowNull: false,
            },
            previous_data: {
                type: DataTypes.TEXT('medium'),
                allowNull: false,
            },
            updated_data: {
                type: DataTypes.TEXT('medium'),
                allowNull: false,
            },
        },
        {
            updatedAt: true,
            createdAt: false,
            freezeTableName: true,
        },
    );

    UserProcess.associate = (models) => {
        UserProcess.belongsTo(models.User, { foreignKey: 'user_id', as: 'users' });
        UserProcess.belongsTo(models.Process, { foreignKey: 'process_id', as: 'processes' });
    };

    return UserProcess;

};
